src/auth/auth.service.ts

```diff
@@ -0,0 +1,115 @@
+import {
+  HttpException,
+  HttpStatus,
+  Injectable,
+  Logger,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { LoginDto } from './dto/login.dto';
+import { UserService } from './user/user.service';
+import {
+  ERROR_MESSAGES,
+  SUCCESS_MESSAGES,
+} from '@app/common/constants/constants';
+import { JwtService } from '@nestjs/jwt';
+import { CreateUserDto } from './user/dto/create-user.dto';
+import { User } from './user/entity/user.model';
+import { Response } from 'express';
+import { TWO_HOURS_FROM_NOW_DATE } from '@app/common/constants/constants';
+import * as bcrypt from 'bcryptjs';
+import { TokenPayload } from 'src/interfaces/TokenPayload';
+import { ConfigService } from '@nestjs/config';
+@Injectable()
+export class AuthService {
+  constructor(
+    private userService: UserService,
+    private jwtService: JwtService,
+    private configService: ConfigService,
+  ) {}
+
+  async login(loginDto: LoginDto, response: Response) {
+    const dbUser = await this.userService.getUser(loginDto.username);
+    if (!dbUser) {
+      Logger.error('User not found');
+      throw new UnauthorizedException(ERROR_MESSAGES.INVALID_CREDENTIALS);
+    }
+    const isPasswordMatching = await bcrypt.compare(
+      loginDto.password,
+      dbUser.password,
+    );
+    if (!isPasswordMatching) {
+      Logger.error('password not matching');
+      throw new UnauthorizedException(ERROR_MESSAGES.INVALID_CREDENTIALS);
+    }
+    const tokenPayload: TokenPayload = {
+      sub: dbUser.id,
+      username: dbUser.username,
+    };
+    const jwtToken = this.jwtService.sign(tokenPayload);
+
+    const expires = new Date();
+    expires.setSeconds(
+      expires.getSeconds() + this.configService.get('JWT_EXPIRATION'),
+    );
+    response.cookie('Authentication', jwtToken, {
+      httpOnly: true,
+      expires: TWO_HOURS_FROM_NOW_DATE,
+    });
+    return jwtToken;
+  }
+
```

Comment:
There's double logic in here. The local strategy is already executed before the login controller, you already returned the user in local strategy when verifying the user you checked that the user exists in the database and properly returned it if it does. 

So why there's a need to recheck if it exists in the DB again and do all of these comparison again?

You need to refactor this function to only sign the user's JWT cookie and return the token. And Also the LoginDTO should be only returning the user itself from the verifyUser in userService.